crates/sui-types/src/base_types.rs

```diff
@@ -1044,13 +1056,17 @@ impl TxContext {
         digest: &TransactionDigest,
         epoch_id: &EpochId,
         epoch_timestamp_ms: u64,
+        gas_price: u64,
+        sponsor: Option<SuiAddress>,
     ) -> Self {
         Self {
             sender: AccountAddress::new(sender.0),
             digest: digest.into_inner().to_vec(),
             epoch: *epoch_id,
             epoch_timestamp_ms,
             ids_created: 0,
+            gas_price,
+            sponsor: sponsor.map(|s| AccountAddress::new(s.0)),
```

Comment:
Would `s.into()` work here (or barring that `AccountAddress::from(s)`)? It should/I would hope it does...

Reply:
done